packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py

```diff
@@ -766,8 +772,10 @@ def do_reasoning_with_retry(
         except Exception as e:
             print(f"Attempt {attempt + 1} failed with error: {e}")
             time.sleep(delay)
+            # join the tool errors with the exception message
+            tool_errors += f"Attempt {attempt + 1} failed with error: {e}\n"
```

Comment:
Why print the error?

```suggestion
            error = f"Attempt {attempt + 1} failed with error: {e}"
            print(error)
            time.sleep(delay)
            # join the tool errors with the exception message
            tool_errors += f"{error}\n"
```